hack/timestamps.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+function timestamps::time_wrapper() {
+    while IFS= read -r line; do
+        printf "%(%T)T: %s\n" "-1" "$line"
+    done
+}
+
+if [[ "${BASH_SOURCE[0]}" == "$0" ]]; then
+    set -o pipefail
```

Comment:
We will have to monitor how this behaves, I am unclear how it will affect the existing scripts in general.

Reply:
return ${PIPESTATUS[0]} worth trying , it should preserve the original retcode, no need to change to pipefail if so, which might change the behavior as you said